### DISCUSSION TITLE
Select is not working with basic Provider?

### DISCUSSION THREAD

**Discussion by MisterLight** [rrousselGit/riverpod#2428] [Q&A] [answered]
I have a provider that store a model like this :

```
  class FormAgeData {
    String? age;
    bool? isApplicable;
  }
  
  final formAgeProvider = Provider.autoDispose((ref)  {
   return  FormAgeData();
  }
```

Im using this provider to store the data that user write on each input of the form, just like this:

```
class AgeInputWidget extends ConsumerWidget{
   Widget build(BuildContext context, WidgetRef ref) {
    final formData = ref.read(formAgeProvider);
   
   return FormField(
    onChange: (value) => formData.age = value;
   )
}

class IsApplicableInputWidget extends ConsumerWidget{
   Widget build(BuildContext context, WidgetRef ref) {
    final formData = ref.read(formAgeProvider);
    final  userAge =  ref.watch(formAgeProvider.select((value) => value.age);

   return CustomSwitch(
    isDisable: () { return userAge >  MAX_AGE})
    onChange: (value) => formData.isApplicable = value;
   )
}
```

Both component are working perfectly, I want to disable the switch of the second component depending on the age, but when I change the age property of the object inside the provider in the AgeInputWidget, the IsApplicableInputWidget widget dont rebuild,  as you can see Im listening to this specific property. I don't understand why isn't working.

Any idea? Is because Im using a Provider instead of a StateProvider?

**Comment by rrousselGit** [comment, marked as answer]
select has nothing to do with it. Your problem is your `onChange` function

There's nothing in your `onChange` that notifies listeners somehow. Either use a Notifier or call ref.notifyListeners in your provider.
Otherwise it's as if nothing happened

**Reply by MisterLight** [reply]
Thank you for the response. I change my code to use a Notifier. I try first at first using  Notifier without any additional method, but looks like the changes inside the object properties doesn't trigger the state change, so I need to use an additional method to force the call to listeners. I don't know if there is a better way of doing this, what do you think?


```
class FormAgeData {
  String? age;
  bool? isApplicable;
}

class FormAgeDataNotifier extends Notifier<FormAgeData> {
  @override
  FormAgeData build() {
    return FormAgeData();
  }

  forceUpdate() {
    ref.notifyListeners();
  }
}

final formAgeProvider =
    NotifierProvider<FormAgeDataNotifier, FormAgeData>(FormAgeDataNotifier.new);
```
